### PR TITLE
Api structural variant rebased

### PR DESCRIPTION
**Pull request by sandertan**
Rebased version of https://github.com/thehyve/cbioportal/pull/24

### CODE REVIEW

**Comment by dionnezaal** [comment]
Have added second sample to test database to make the MyBatisRepository test `fetchStructuralVariantsWithSampleIdentifiers` indeed check if the sampleId is filtered.

File: web/src/main/java/org/cbioportal/web/StructuralVariantController.java [hunk 1]
```diff
@@ -58,26 +58,26 @@
...
             }
-            structuralVariantList = structuralVariantService.fetchStructuralVariants(structuralVariantFilter.getGeneticProfileStableIds(), structuralVariantFilter.getHugoGeneSymbols(), studyIds, sampleIds);
+            structuralVariantList = structuralVariantService.fetchStructuralVariants(molecularProfileIds, structuralVariantFilter.getHugoGeneSymbols(), sampleIds);
             
         } else {
-            List<String> studyIds = new ArrayList<>();
+            List<String> molecularProfileIds = new ArrayList<>();
             List<String> sampleIds = new ArrayList<>();
-            structuralVariantList = structuralVariantService.fetchStructuralVariants(structuralVariantFilter.getGeneticProfileStableIds(), structuralVariantFilter.getHugoGeneSymbols(), studyIds, sampleIds);
+            structuralVariantList = structuralVariantService.fetchStructuralVariants(molecularProfileIds, structuralVariantFilter.getHugoGeneSymbols(), sampleIds);
```
Comment (rnugraha) [inline web/src/main/java/org/cbioportal/web/StructuralVariantController.java]: Why using empty array? Shouldn't this use structuralVariantFilter.getMolecularProfileIds()?

File: web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java [hunk 2]
```diff
@@ -217,20 +217,20 @@ private StructuralVariantFilter createStructuralVariantFilter( ) {
...
+        sampleMolecularIdentifier1.setMolecularProfileId(TEST_GENETIC_PROFILE_STABLE_ID_1);
+        sampleMolecularIdentifierList.add(sampleMolecularIdentifier1);
 
-        structuralVariantFilter.setGeneticProfileStableIds(geneticProfileStableIds);
+//        structuralVariantFilter.setMolecularProfileIds(molecularProfileIds);
         structuralVariantFilter.setHugoGeneSymbols(hugoGeneSymbols);
-        structuralVariantFilter.setSampleIdentifiers(sampleIdentifierList);
+        structuralVariantFilter.setSampleMolecularIdentifiers(sampleMolecularIdentifierList);
         return structuralVariantFilter;
     }
```
Comment (rnugraha) [inline web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java]: Please provide test with the case where filter will be `molecularProfileIds` and `hugeGeneSymbols`. Even better provide failing test.

File: web/src/main/java/org/cbioportal/web/StructuralVariantController.java (hunk 1 above)
Comment (dionnezaal) [reply web/src/main/java/org/cbioportal/web/StructuralVariantController.java]: You are correct! This is a mistake, should indeed be structuralVariantFilter.getMolecularProfileIds().

**Comment by dionnezaal** [comment]
@rnugraha Have fixed mistake in controller with molecularProfileId. Also added two more test to the controller, one for when molecularProfileIds are given as input, and failing test when input contains both MolecularProfileId and SampleMolecularIdentifiers.

**Comment by rnugraha** [comment]
@dionnezaal Checks are still failing, could you fix them? Maybe this is also due to the root branch (Sander's)

@sandertan @pieterlukasse I would really appreciate merging this PR ASAP. Do we have any concerns that hold this PR to be merged?

**Comment by pieterlukasse** [comment]
@rnugraha It will have to be reviewed before it gets merged. Why did we move away from the PR where review had started? Just checking...I can continue review here.

**Comment by dionnezaal** [comment]
@rnugraha Will work on it as soon as possible

@pieterlukasse This version was rebased by sander from https://github.com/thehyve/cbioportal/pull/24 when I had a day off, afterwards continued on this rebased version.

File: model/src/main/java/org/cbioportal/model/StructuralVariant.java [hunk 3]
```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (c) 2017 The Hyve B.V.
```
Comment (rnugraha) [inline model/src/main/java/org/cbioportal/model/StructuralVariant.java]: 2018

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.cbioportal.persistence.mybatis;
```
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java]: Add copyright

File: persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml [hunk 5]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/StructuralVariantMapper.xml]: Add copyright

File: persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
+package org.cbioportal.persistence.mybatis;
```
Comment (rnugraha) [inline persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java]: Add copyright

File: service/src/main/java/org/cbioportal/service/StructuralVariantService.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package org.cbioportal.service;
```
Comment (rnugraha) [inline service/src/main/java/org/cbioportal/service/StructuralVariantService.java]: Add copyright

File: service/src/main/java/org/cbioportal/service/StructuralVariantService.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.cbioportal.model.StructuralVariant;
+
+import java.util.List;
+
+public interface StructuralVariantService {
+    
+    List<StructuralVariant> fetchStructuralVariants(List<String> molecularProfileIds, 
+            List<Integer> entrezGeneIds, List<String> sampleIds);
+}
```
Comment (rnugraha) [inline service/src/main/java/org/cbioportal/service/StructuralVariantService.java:34]: Add new line

File: service/src/main/java/org/cbioportal/service/impl/StructuralVariantServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
+package org.cbioportal.service.impl;
```
Comment (rnugraha) [inline service/src/main/java/org/cbioportal/service/impl/StructuralVariantServiceImpl.java]: Add copyright

File: service/src/test/java/org/cbioportal/service/impl/StructuralVariantServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
+package org.cbioportal.service.impl;
```
Comment (rnugraha) [inline service/src/test/java/org/cbioportal/service/impl/StructuralVariantServiceImplTest.java]: Add copyright

File: web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
+package org.cbioportal.web.parameter;
```
Comment (rnugraha) [inline web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java]: Add copyright

File: web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    
+    public List<SampleMolecularIdentifier> getSampleMolecularIdentifiers(){
+        return sampleMolecularIdentifiers;
+    }
+    
+    public void setSampleMolecularIdentifiers(List<SampleMolecularIdentifier> sampleMolecularIdentifiers) {
+        this.sampleMolecularIdentifiers = sampleMolecularIdentifiers;
+    }
+}
```
Comment (rnugraha) [inline web/src/main/java/org/cbioportal/web/parameter/StructuralVariantFilter.java]: Add new line

File: web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java [hunk 13]
```diff
@@ -0,0 +1,366 @@
+package org.cbioportal.web;
```
Comment (rnugraha) [inline web/src/test/java/org/cbioportal/web/StructuralVariantControllerTest.java]: Add copyright

File: persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+        molecularProfileIds.add("study_tcga_pub_sv");
+        entrezGeneIds.add(57670);
+
+        List<StructuralVariant> result = 
+                structuralVariantMyBatisRepository.fetchStructuralVariants(molecularProfileIds, 
+                        entrezGeneIds, sampleIds);
+
+        Assert.assertEquals(2,  result.size());
+        StructuralVariant structuralVariantFirstResult = result.get(0);
+        Assert.assertEquals("study_tcga_pub_sv", structuralVariantFirstResult.getMolecularProfileId());
```
Comment (rnugraha) [inline persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepositoryTest.java]: Why comparing each attributes one by one? Can't we just assert the whole object?

**Review by rnugraha — CHANGES_REQUESTED** [review]

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+public class StructuralVariantMyBatisRepository implements StructuralVariantRepository {
+
+    @Autowired
+    private StructuralVariantMapper structuralVariantmapper;
+    
+    @Override
+    public List<StructuralVariant> fetchStructuralVariants(List<String> molecularProfileIds, 
+            List<Integer> entrezGeneIds, List<String> sampleIds) {
+
+        return structuralVariantmapper.fetchStructuralVariants(molecularProfileIds,
```
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java]: structuralVariantmapper pls changed to structuralVariantMapper. Pls maintain the camel case code styling.

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class StructuralVariantMyBatisRepository implements StructuralVariantRepository {
+
+    @Autowired
+    private StructuralVariantMapper structuralVariantmapper;
```
Comment (rnugraha) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/StructuralVariantMyBatisRepository.java]: structuralVariantmapper pls changed to structuralVariantMapper. Pls maintain the camel case code styling.

**Review by rnugraha — CHANGES_REQUESTED** [review]

**Review by oplantalech — APPROVED** [review]
Good job, I've revised all the code and it looks good.

**Comment by pieterlukasse** [comment]
@oplantalech please squash the commits.

**Comment by oplantalech** [comment]
@pieterlukasse Done

**Review by rnugraha — APPROVED** [review]

**Pull request merged**